### PR TITLE
feat: deprecate unexpected config option value types

### PR DESCRIPTION
**Pull request by Hadis-Fard**
Added deprecation for boolean config types, as the changes in #692 are breaking change. Left the deprecation around encrypt to #695

### CODE REVIEW

File: src/connection.js [hunk 1]
```diff
@@ -259,54 +343,79 @@ class Connection extends EventEmitter {
...
       if (config.options.multiSubnetFailover != undefined) {
+        deprecateNonBooleanConfigValue('options.multiSubnetFailover', config.options.multiSubnetFailover);
+
         this.config.options.multiSubnetFailover = !!config.options.multiSubnetFailover;
       }
 
-      if (config.options.packetSize) {
+      if (config.options.packetSize != undefined) {
+        if (config.options.packetSize <= 0 || config.options.packetSize >= 65536) {
+          throw new RangeError('Port must be > 0 and < 65536');
```
Comment (Hadis-Fard) [inline src/connection.js]: Error is thrown for port number should be packetsize 😉 , also this is a breaking change, it's better to include it in breaking changes PR

File: src/connection.js [hunk 2]
```diff
@@ -116,68 +162,97 @@ class Connection extends EventEmitter {
 
         this.config.options.abortTransactionOnError = config.options.abortTransactionOnError;
       }
+      deprecateNullFallbackToDefaultConfigValue('options.abortTransactionOnError', config.options.abortTransactionOnError);
```
[the remark this answers is not in the document]
Comment (Hadis-Fard) [reply src/connection.js]: 👍 to `deprecateNullFallbackToDefaultConfigValue`

File: src/connection.js (hunk 1 above)
Comment (arthurschreiber) [reply src/connection.js]: I wouldn't _quite_ classify this as a breaking change (passing any other value here would make the connection fail), but you're right, let's just put it into the breaking changes pull request. 👍

File: src/connection.js [hunk 3]
```diff
@@ -116,68 +162,97 @@ class Connection extends EventEmitter {
...
       if (config.options.database != undefined) {
+        deprecateNonStringConfigValue('options.database', config.options.database);
         this.config.options.database = config.options.database;
       }
+      deprecateNullConfigValue('options.database', config.options.database);
 
       if (config.options.datefirst) {
         if (config.options.datefirst < 1 || config.options.datefirst > 7) {
-          throw new RangeError('DateFirst should be >= 1 and <= 7');
+          throw new RangeError('options.datefirst must be a number >= 1 and <= 7');
```
[the remark this answers is not in the document]
Comment (arthurschreiber) [reply src/connection.js]: Ugh. Yeah, this kinda depends on whether we think error _messages_ are something users can or will depend on. I like to believe it is not 😅 - but I might be wrong.

I agree on pulling this out into the other pull request for breaking changes.

File: src/connection.js [hunk 4]
```diff
@@ -116,68 +162,97 @@ class Connection extends EventEmitter {
...
       if (config.options.cancelTimeout != undefined) {
         this.config.options.cancelTimeout = config.options.cancelTimeout;
       }
+      deprecateNullConfigValue('options.cancelTimeout', config.options.cancelTimeout);
 
-      if (config.options.columnNameReplacer) {
+      if (config.options.columnNameReplacer !== undefined) {
         if (typeof config.options.columnNameReplacer !== 'function') {
-          throw new TypeError('options.columnNameReplacer must be a function or null.');
+          throw new TypeError('options.columnNameReplacer must be a function.');
```
[the remark this answers is not in the document]
Comment (arthurschreiber) [reply src/connection.js]: Yup! See my other responses further below! 👍

File: src/connection.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (arthurschreiber) [reply src/connection.js]: > Would be good to add TODO to check for null values in getInitialSql before pushing the value into options

Yeah. Would you like to prepare a pull request for that?

File: src/connection.js [hunk 5]
```diff
@@ -47,12 +79,26 @@ class Connection extends EventEmitter {
...
+    }
+    deprecateNullConfigValue('domain', config.domain);
+
+    if (config.userName != undefined) {
+      deprecateNonStringConfigValue('userName', config.userName);
+    }
+    deprecateNullConfigValue('userName', config.userName);
+
+    if (config.password != undefined) {
+      deprecateNonStringConfigValue('userName', config.password);
```
[the remark this answers is not in the document]
Comment (arthurschreiber) [reply src/connection.js]: Ugh - thanks! 🙇

**Comment by Hadis-Fard** [comment]
LGTM

**Comment by arthurschreiber** [comment]
@Hadis-Fard Sorry, I was mucking around with this branch (as can be seen by the force pushes I made), and I believe that caused you a bunch of trouble. 😓 I'll refrain from doing that in the future, as it just causes pain without a lot of gain. 🙇

**Comment by Hadis-Fard** [comment]
@arthurschreiber I appreciate your contribution to this branch, and I just had to merge two remotes so it wasn't a pain 😺

**Pull request merged**